src/sql/db_connection_pool/dbconnection/mysqlconn.rs

```diff
@@ -72,11 +83,63 @@ impl<'a> AsyncDbConnection<Conn, &'a (dyn ToValue + Sync)> for MySQLConnection {
             .boxed()
             .context(super::UnableToGetSchemaSnafu)?;
 
-        let rec = rows_to_arrow(&rows)
+        if !rows.is_empty() {
+            let rec = rows_to_arrow(&rows)
+                .boxed()
+                .context(super::UnableToGetSchemaSnafu)?;
+            return Ok(rec.schema());
+        }
+
+        // we don't use SELECT COLUMN_NAME, DATA_TYPE FROM INFORMATION_SCHEMA.COLUMNS WHERE TABLE_NAME = '{}' AND TABLE_SCHEMA = '{}'
+        // as table_reference don't always have schema specified so we need to extract schema/db name from connection properties
+        // to ensure we are querying information for correct table
+        let metadata_query = format!("SHOW COLUMNS FROM {}", table_reference.to_quoted_string());
```

Comment:
Let's refactor this into a separate method, similar to `rows_to_arrow` we can call. Maybe `metadata_rows_to_schema`?

Reply:
Updated - moved to `columns_meta_to_schema` fn